pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/stats/BytesColumnPredIndexStatsCollector.java

```diff
@@ -59,11 +59,11 @@ public BytesColumnPredIndexStatsCollector(String column, StatsCollectorConfig st
   private void collectEntry(Object entry, ObjectSet<ByteArray> set) {
     if (entry instanceof Object[]) {
       for (final Object e : (Object[]) entry) {
-        byte[] value = (byte[]) e;
-        set.add(new ByteArray(value));
+        ByteArray value = ((ByteArray) e).clone(); // Underlying byte[] may be re-cycled.
```

Comment:
After PlainFieldExtractor, this value should always be byte[]？

Reply:
No, If you look at the stats collector code, it expects comparable, hence the ByteArray wrapper.